projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/KubernetesService.java

```diff
@@ -1314,14 +1318,36 @@ public void saveSecretData(String name, Map<String, byte[]> data) throws ApiExce
         try {
             nsSecret = api.replaceNamespacedSecret(name, this.namespace, secret, null, null, null);
         } catch (ApiException e) {
+            log.warn("Error while trying to save K8S secret", e);
             if (e.getCode() == 404) {
-                log.debug("Secret {} does not exits. Creating ...");
+                log.debug("Secret does not exist. Creating ...");
```

Comment:
Let's mention the name of the secret . It was missed before . 

I think it's usually better  to have  less generic log messages so that user can see what is going on . Printing secret name is kind important here. I would print it in the warn above as well and all logs of this method.

Reply:
Changed.